### PR TITLE
Add database unit tests

### PR DESCRIPTION
**Pull request by raghavi101**
This PR :

Adds unit tests (<file_name>_test.go) for the following db files located in backend-shared/config/db 
- managedenvironment.go
- kubernetesresourcetodbresourcemapping.go
- gitopsengineinstance.go

It is a part of [GITOPSRVCE-100](https://issues.redhat.com/browse/GITOPSRVCE-100)

Needs debugging : 
```
Error Trace:    kubernetesresourcetodbresourcemapping_test.go:103
                Error:          Received unexpected error:
                                error on deleting operation: pg: model=KubernetesToDBResourceMapping does not have primary keys
```

### CODE REVIEW

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, thanks @raghava101!

I have pushed a fix to your branch re:  the errors you were seeing (the pointer one, and the primary key one). You will need to do `git pull` on your branch to pick up the changes.

A few items that jumped out at me as I was reviewing:

#### 1) As I was reviewing, I realized an additional thing that we should test:
- In the Get test, call Get with a primary key that doesn't exist: ensure that an error is returned, and that the IsResultNotFoundError(error) is true.
- Replace assert.Equal with assert.ObjectsAreEqualValues, when comparing the before and after calls of a get method 

I've added this to the story criteria of the story. We should do this for all the `Get*` functions in the PR.


#### 2) Also: 

Can you add new code to testSetup(), that will delete any KubernetesResourceToDBResourceMapping DB table entries that have a primary key that begins with "test-"?

In order to implement this, you may need to add a new 'UnsafeListAllKubernetesResourceToDBResourceMapping' function (see similar UnsafeListAll) examples in the code for details.

**Comment by jgwest** [comment]
FYI @raghavi101 On another PR, I switched the code to require Go 1.17, so you may need to:
- update your system's Go version to 1.17
- rebase your branch onto the main branch
- run `go mod tidy` on backend-shared
- push any updates to the PR

**Review by jgwest — CHANGES_REQUESTED** [review]

File: backend-shared/config/db/gitopsengineinstance_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		Namespace_name:          "test-fake-namespace",
+		Namespace_uid:           "test-fake-namespace-2",
+		EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+	}
+
+	err = dbq.GetGitopsEngineInstanceById(ctx, &gitopsEngineInstanceget)
+	if !assert.NoError(t, err) {
+		return
+	}
+	assert.Equal(t, gitopsEngineInstanceput.Namespace_name, gitopsEngineInstanceget.Namespace_name)
```
Comment (jgwest) [inline backend-shared/config/db/gitopsengineinstance_test.go:126]: As above, should do assert.Equal on the full object.

File: backend-shared/config/db/kubernetesresourcetodbresourcemapping_test.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		KubernetesResourceUID:  "test_resource_uid",
+		DBRelationType:         "test_relation_type",
+		DBRelationKey:          "test_relation_key",
+	}
+
+	err = dbq.GetDBResourceMappingForKubernetesResource(ctx, &kubernetesToDBResourceMappingget)
+	if !assert.NoError(t, err) {
+		return
+	}
+	assert.Equal(t, kubernetesToDBResourceMappingpost.KubernetesResourceUID, kubernetesToDBResourceMappingget.KubernetesResourceUID)
```
Comment (jgwest) [inline backend-shared/config/db/kubernetesresourcetodbresourcemapping_test.go:76]: As above.

File: backend-shared/config/db/managedenvironment_test.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	getmanagedEnvironment := ManagedEnvironment{
+		Managedenvironment_id: "test-managed-env-4",
+		Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+		Name:                  "my env101",
+	}
+	get := dbq.GetManagedEnvironmentById(ctx, &getmanagedEnvironment)
+	if !assert.NoError(t, get) {
+		return
+	}
+	assert.Equal(t, managedEnvironment.Name, getmanagedEnvironment.Name)
```
Comment (jgwest) [inline backend-shared/config/db/managedenvironment_test.go:101]: As above, should compare full object.

**Comment by raghavi101** [comment]
Thanks Jonathan for your review! Committed all the changes that were pointed out in the comments. PTAL

File: backend-shared/config/db/managedenvironment_test.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+	managedEnvironmentNotExist := ManagedEnvironment{
+		Managedenvironment_id: "test-managed-env-4-not-exist",
+		Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+		Name:                  "my env101-not-exist",
+	}
+
+	err = dbq.GetManagedEnvironmentById(ctx, &managedEnvironmentNotExist)
+	if assert.Error(t, err) {
+		if !IsResultNotFoundError(err) {
```
Comment (jgwest) [inline backend-shared/config/db/managedenvironment_test.go:113]: Asserts as above.

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, just one more tweak (add an `assert` across a few methods) then good to go.

Oh and make sure you do this, so that the PR checks start passing: https://github.com/redhat-appstudio/managed-gitops/pull/74#issuecomment-1061664270

**Comment by raghavi101** [comment]
Done, pushed requested changes. (Glad the checks passed, thanks Jonathan!)

**Review by samyak-jn — COMMENTED** [review]
Hi!
I was going through the PR and on the first quick look I found out a `.DS_Store` file is also committed to it, I suspect it might be a result of doing `git add .` or is it expected? If not, clear the file from the PR.

**Comment by raghavi101** [comment]
Hi Samyak! Thanks, I made the reviewed changes

File: backend-shared/config/db/gitopsengineinstance_test.go [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+	gitopsEngineInstanceNotExist := GitopsEngineInstance{
+		Gitopsengineinstance_id: "test-fake-engine-instance-id-not-exist",
+		Namespace_name:          "test-fake-namespace-not-exist",
+		Namespace_uid:           "test-fake-namespace-1-not-exist",
+		EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+	}
+
+	err = dbq.GetGitopsEngineInstanceById(ctx, &gitopsEngineInstanceNotExist)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/gitopsengineinstance_test.go:78]: This should be `if !assert.True(`, I believe.

File: backend-shared/config/db/gitopsengineinstance_test.go [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+	rowsAffected, err = dbq.DeleteGitopsEngineClusterById(ctx, gitopsEngineCluster.Gitopsenginecluster_id)
+	assert.NoError(t, err)
+	assert.Equal(t, rowsAffected, 1)
+
+	rowsAffected, err = dbq.DeleteClusterCredentialsById(ctx, clusterCredentials.Clustercredentials_cred_id)
+	assert.NoError(t, err)
+	assert.Equal(t, rowsAffected, 1)
+
+	err = dbq.GetGitopsEngineInstanceById(ctx, &gitopsEngineInstance)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/gitopsengineinstance_test.go:202]: Should be `if !assert.True(`

File: backend-shared/config/db/gitopsengineinstance_test.go [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+		return
+	}
+	err = dbq.GetGitopsEngineClusterById(ctx, &gitopsEngineCluster)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/gitopsengineinstance_test.go:206]: Should be `if !assert.True(`

File: backend-shared/config/db/gitopsengineinstance_test.go [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+		return
+	}
+	err = dbq.GetClusterCredentialsById(ctx, &clusterCredentials)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/gitopsengineinstance_test.go:210]: Should be `if !assert.True(`

File: backend-shared/config/db/kubernetesresourcetodbresourcemapping_test.go [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+	kubernetesToDBResourceMappingNotExist := KubernetesToDBResourceMapping{
+		KubernetesResourceType: "test-resource_2_not_exist",
+		KubernetesResourceUID:  "test-resource_uid_not_exist",
+		DBRelationType:         "test-relation_type_not_exist",
+		DBRelationKey:          "test-relation_key_not_exist",
+	}
+	//check for inexistent primary key
+	err = dbq.GetDBResourceMappingForKubernetesResource(ctx, &kubernetesToDBResourceMappingNotExist)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/kubernetesresourcetodbresourcemapping_test.go:86]: Should be `if !assert.True(`

File: backend-shared/config/db/managedenvironment_test.go [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+	//check for inexistent primary key
+
+	managedEnvironmentNotExist := ManagedEnvironment{
+		Managedenvironment_id: "test-managed-env-4-not-exist",
+		Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+		Name:                  "my env101-not-exist",
+	}
+
+	err = dbq.GetManagedEnvironmentById(ctx, &managedEnvironmentNotExist)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/managedenvironment_test.go:112]: Should be `if !assert.True(`

File: backend-shared/config/db/managedenvironment_test.go [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+	rowsAffected, err := dbq.DeleteManagedEnvironmentById(ctx, managedEnvironment.Managedenvironment_id)
+	assert.NoError(t, err)
+	assert.Equal(t, rowsAffected, 1)
+
+	rowsAffected, err = dbq.DeleteClusterCredentialsById(ctx, clusterCredentials.Clustercredentials_cred_id)
+	assert.NoError(t, err)
+	assert.Equal(t, rowsAffected, 1)
+
+	err = dbq.GetManagedEnvironmentById(ctx, &managedEnvironment)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/managedenvironment_test.go:161]: Should be `if !assert.True(`

File: backend-shared/config/db/managedenvironment_test.go [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+		return
+	}
+
+	err = dbq.GetClusterCredentialsById(ctx, &clusterCredentials)
+	if assert.True(t, IsResultNotFoundError(err)) {
```
Comment (jgwest) [inline backend-shared/config/db/managedenvironment_test.go:166]: Should be `if !assert.True(`

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks like we need to update a number of the`if assert.True(` calls: the `assert.*` functions return true  on success, so we should only return on failure of these statements.

**Pull request merged**